---
uint16 id
uint16 size